packages/syft/src/syft/core/node/common/node.py

```diff
@@ -114,25 +120,58 @@ def __init__(
             name=name, network=network, domain=domain, device=device, vm=vm
         )
 
+        # TableBase is the base class from which all ORM classes must inherit
+        # If one isn't provided then we can simply make one.
+        if TableBase is None:
+            TableBase = declarative_base()
+
+        # If not provided a session connecting us to the database, let's just
+        # initialize a database in memory
+        if db_session is None:
+
+            # If a DB engine isn't provided then
+            if engine is None:
+                engine = create_engine("sqlite:///:memory:", echo=False)
```

Comment:
For example here, we are making a hard commit that we want sqlite, but maybe in two months we would like to experiment with Redis. It would be nice to be able to receive an additional optional argument in the constructor like `Memory Storage Backend` in which we pick multiple implementations (that's why we have the storage interface). Nice work, but we should totally wrap this up into something else, the node shouldn't be aware of this.

Reply:
Ah so SQLAlchemy can work with many sql backends - and in this case this is just one of the ways to ask for an in-memory SQL store.